roles/splunk/defaults/main.yml

```diff
@@ -31,6 +31,12 @@ splunk_admin_username: admin
 splunk_admin_password: undefined # Use ansible-vault encrypt_string, e.g. ansible-vault encrypt_string --ask-vault-pass 'var_value_to_encrypt' --name 'var_name'
 splunk_configure_secret: false # If set to true, you need to update files/splunk.secret
 splunk_secret_file: splunk.secret # Used to specify your splunk.secret filename(s), files should be placed in the "files" folder of the role
+configure_firewall: false # Whether or not to configure the firewall service on your machine, if set to true, opens firewall ports using UFW (default) or Firewalld depending on OS
+splunk_firewall_ports: # List of ports to allow through local firewall in dict form
```

Comment:
I think it would be nice to have vars for the other ports as well so it can be customized, Something like `splunkweb_port` and `splunkhec_port` etc.

Reply:
Since I'm a novice at variable precedence, where should these be defined? I'm wishing they could be in their own separate file in `vars/Firewall.yml` but I don't know if that takes precedence over the `group_vars` - if it does, it wouldn't make it very useful, which leaves us the only option of defining them inline in `defaults/main.yml`. What do you think?

Reply:
Here is a nice list of the precedences.
https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#understanding-variable-precedence
I would put it in `defaults/main.yml`, and then uverride it per group in `group_vars` or per host in `host_vars`.

If you use `include_vars` in a task to include `Firewall.yml`, it'll override whatever you set in the var files.

Reply:
Alright for sure, it's just going to clog up the defaults file. As long as you're okay with that I'll proceed.

Reply:
Done in https://github.com/splunk/ansible-role-for-splunk/pull/157/commits/f93c8a18104136f20458ce214e3770068b9d1f80